unit_tests/testMeshTransfer/testMeshTransfer.f90

```diff
@@ -0,0 +1,1040 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+PROGRAM testMeshTransfer
+#include "UnitTest.h"
+  USE UnitTest
+  USE IntrType
+  USE Strings
+  USE ParameterLists
+  USE MeshTransfer
+
+  IMPLICIT NONE
+!
+!Check the timer resolution
+  CREATE_TEST('Mesh Transfer')
+
+  REGISTER_SUBTEST('Test LP PointVal',TestLPPointVal)
+  REGISTER_SUBTEST('Test LP Integral',TestLPIntegral)
+  REGISTER_SUBTEST('Test ZP PointVal',TestZPPointVal)
+  REGISTER_SUBTEST('Test ZP Integral',TestZPIntegral)
+  REGISTER_SUBTEST('Test 1DBase MeshTransfer',Test1DBase)
+  REGISTER_SUBTEST('Test 1D setup',Test1Dsetup)
+  REGISTER_SUBTEST('Test 1DCart MeshTransfer',Test1DCart)
+  REGISTER_SUBTEST('Test 1DCyl MeshTransfer',Test1DCyl)
+
+  FINALIZE_TEST()
+!
+!
+!===============================================================================
+  CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> Using 1DCart but testing only functions on the 1Dbase class.
+    SUBROUTINE Test1DBase()
+      TYPE(MeshTransfer_1DCart) :: testMT
+      TYPE(ParamType) :: testPL
+      INTEGER(SIK) :: i
+      REAL(SRK) :: mesh(6), tmpin(5),mesh_out(11)
+      REAL(SRK),ALLOCATABLE :: tmpout(:)
+
+      DO i=0,5
+        mesh(i+1)=SQRT(REAL(i,SRK)/5.0_SRK)
+      ENDDO
+      DO i=0,10
+        mesh_out(i+1)=SQRT(REAL(i,SRK)/10.0_SRK)
+      ENDDO
+
+      CALL testPL%add('MeshTransfer->map_in','VOLUME')
+      CALL testPL%add('MeshTransfer->map_out','POINT')
+      CALL testPL%add('MeshTransfer->volumemesh_in',mesh)
+      CALL testPL%add('MeshTransfer->pointmesh_out',mesh_out)
+      CALL testPL%add('MeshTransfer->poly_transfer',4)
+
+      COMPONENT_TEST('%init_base')
+      CALL testMT%init_base(testPL)
+
+      ASSERT_EQ(testMT%MapType_in,3,'map_in')
```

Comment:
should we use the enumerations when checking MapType_in and MapType_out?

Reply:
I didn't because I hand't made them public.  This was mostly because the test is the only reason to.